static/app/views/insights/common/components/chartPanel.tsx

```diff
@@ -1,21 +1,74 @@
+import {useMemo} from 'react';
 import styled from '@emotion/styled';
 
 import {openInsightChartModal} from 'sentry/actionCreators/modal';
 import {Button} from 'sentry/components/button';
+import {DropdownMenu} from 'sentry/components/dropdownMenu';
 import Panel from 'sentry/components/panels/panel';
-import {IconExpand} from 'sentry/icons';
+import {IconEllipsis, IconExpand} from 'sentry/icons';
 import {t} from 'sentry/locale';
 import {space} from 'sentry/styles/space';
+import useOrganization from 'sentry/utils/useOrganization';
+import usePageFilters from 'sentry/utils/usePageFilters';
+import useProjects from 'sentry/utils/useProjects';
+import {getAlertsUrl} from 'sentry/views/insights/common/utils/getAlertsUrl';
+import {hasInsightsAlerts} from 'sentry/views/insights/common/utils/hasInsightsAlerts';
 import {Subtitle} from 'sentry/views/performance/landing/widgets/widgets/singleFieldAreaWidget';
 
+export type AlertConfig = {
+  aggregate: string;
+  name?: string;
+  query?: string;
+};
+
 type Props = {
   children: React.ReactNode;
+  alertConfigs?: AlertConfig[];
   button?: JSX.Element;
   subtitle?: React.ReactNode;
   title?: React.ReactNode;
 };
 
-export default function ChartPanel({title, children, button, subtitle}: Props) {
+export default function ChartPanel({
+  title,
+  children,
+  button,
+  subtitle,
+  alertConfigs,
+}: Props) {
+  const organization = useOrganization();
+  const {selection} = usePageFilters();
+  const {projects} = useProjects();
+  const project = useMemo(
+    () =>
+      selection.projects.length === 1
+        ? projects.find(p => p.id === `${selection.projects[0]}`)
+        : undefined,
+    [projects, selection.projects]
+  );
+  const alertsUrls =
+    alertConfigs?.map(alertConfig => {
+      // Alerts only support single project selection
+      const singleProject = selection.projects.length === 1 && project;
+      const alertsUrl =
+        alertConfig && singleProject
+          ? getAlertsUrl({project, ...alertConfig})
+          : undefined;
+      const name = alertConfig.name ?? 'Create Alert';
+      const disabled = !singleProject;
+      return {
+        key: name,
+        label: name,
+        to: alertsUrl,
+        disabled,
+        tooltip: disabled
+          ? t('Alerts are only available for a single project')
```

Comment:
Should we also mention what they should do to enable alerts? I.e "please select only one project", or do we think this message is enough to imply that?

Reply:
Updated the tooltip message!